source/TimePicker.js

```diff
@@ -53,14 +53,67 @@
 		published: {
 			/**
 			* The meridiem text to display if [meridiemEnable]{@link moon.TimePicker#meridiemEnable}
-			* is `true`. The first item is used if the `hour` is less than `11`; otherwise, the
+			* is `true`. The first item is used if the `hour` is less than `12`; otherwise, the
 			* second is used.
 			*
 			* @type {String[]}
-			* @default ['AM','PM']
+			* @default [{name: 'AM', start: '00:00', end: '11:59'}, {name: 'PM', start: '12:00', end: '23:59'}]
 			* @public
 			*/
-			meridiems: ['AM','PM']
+			meridiems: [{name: 'AM', start: '00:00', end: '11:59'}, {name: 'PM', start: '12:00', end: '23:59'}]
+		},
+
+		/**
+		* @private
+		*/
+		initILib: function() {
+			// Get localized meridiem values
+			var fmtParams = {
+				template: 'a',
+				useNative: false,
+				timezone: 'local'
+			};
+
+			if (this.locale) fmtParams.locale = this.locale;
+
+			var merFormatter = new ilib.DateFmt(fmtParams);
+			this.meridiems = merFormatter.getMeridiemsRange(fmtParams);
+		},
+
+		/**
+		* @private
+		*/
+		create: function() {
+			if (typeof ilib !== 'undefined') this.initILib();
+			this.max = this.meridiems.length - 1;
+			this.inherited(arguments);
+		},
+
+		/**
+		* Set meridiem value based on given hour and minute
+		*
+		* @param  {Number} hour - hour between 0 to 23
+		* @param  {Number} minute - minute between 0 to 59
+		*
+		* @public
+		*/
+		setValueByTime: function (hour, minute) {
+			var meridiems = this.meridiems,
+				start, end, time;
+			for (var i = 0; i < meridiems.length; i++) {
+				start = parseInt(meridiems[i]['start'].substring(0,2) + meridiems[i]['start'].substring(3,5), 10);
+				end = parseInt(meridiems[i]['end'].substring(0,2) + meridiems[i]['end'].substring(3,5), 10);
+				time = hour * 100 + minute;
+
+				if ( start <= time && time <= end) {
+					if (this.value != null) {
+						this.set('value', i);
+					} else {
+						//at initial time, we don't want to trigger meridiemPickerChanged method
+						this.value = i;
```

Comment:
Instead of doing this, what do you think about using a call to silence/unsilence, similar to what is done in `moon.DatePicker`? https://github.com/enyojs/moonstone/blob/2.5-upkeep/source/DatePicker.js#L186.